### ISSUE TITLE
GestureManager does not behave like VRChat when GestureLeft & GestureRight are treated as floats in animator

### ISSUE THREAD

**Issue by Antarcticsiclepop** [BlackStartx/VRC-Gesture-Manager#48]
I have my `GestureLeft` & `GestureRight` parameters as floats in my FX Controller and works in VRChat, does not work on unity. GestureManager still treats them as `ints` somehow so my blendtrees don't work.

**Comment by BlackStartx** [comment]
Hi Antarcticsiclepop~ ^-^

I'm working on the parameter coercion feature and it's currently functional in the Gesture Manager 3.8.9 version that is currently in pre-release!

I'm also looking for feedback to ensure I don't have messed up stuff so if you wanna try it out you can download it from here: [Gesture Manager 3.8.9](https://github.com/BlackStartx/VRC-Gesture-Manager/releases/tag/v3.8.9)

Sorry for the issue~
I hope it's correctly fixed in 3.8.9, if you wish and need any help on how to install that version feel free to text me here!

- BlackStartx

**Comment by Antarcticsiclepop** [comment]
Thank you. 🙏🏻

**Comment by Antarcticsiclepop** [comment]
> Hi Antarcticsiclepop~ ^-^
> 
> I'm working on the parameter coercion feature and it's currently functional in the Gesture Manager 3.8.9 version that is currently in pre-release!
> 
> I'm also looking for feedback to ensure I don't have messed up stuff so if you wanna try it out you can download it from here: [Gesture Manager 3.8.9](https://github.com/BlackStartx/VRC-Gesture-Manager/releases/tag/v3.8.9)
> 
> Sorry for the issue~ I hope it's correctly fixed in 3.8.9, if you wish and need any help on how to install that version feel free to text me here!
> 
> * BlackStartx

There seems to be an issue with animated parameters, there values don't show up in the Debug/Avatar/Parameters tab in Gesture Manager. They seem to be changing since stuff is happening on the Scene view, but they all stay as zero in Gesture Manager.

**Comment by BlackStartx** [comment]
>There seems to be an issue with animated parameters, the values don't show up in the Debug/Avatar/Parameters tab in Gesture Manager

Animated parameters? Do you mean the the ones controlled by state behaviours like parameter drivers and others like that?

**Comment by Antarcticsiclepop** [comment]
> > There seems to be an issue with animated parameters, the values don't show up in the Debug/Avatar/Parameters tab in Gesture Manager
> 
> Animated parameters? Do you mean the the ones controlled by state behaviours like parameter drivers and others like that?

The ones controlled by animations.

**Comment by BlackStartx** [comment]
I see, I got what you're talking about~ 🌺

Those type of parameter update have never been properly implemented so it may require some time to make it debug-friendly in 
a new version, but I will work on it! ^-^

I guess the other type of problem is solved, right?
'**GestureLeft**' & '**GestureRight**' (_as well as all the other coercion_) are now treated correctly in the simulation?

**Comment by Antarcticsiclepop** [comment]
> I see, I got what you're talking about~ 🌺
> 
> Those type of parameter update have never been properly implemented so it may require some time to make it debug-friendly in a new version, but I will work on it! ^-^
> 
> I guess the other type of problem is solved, right? '**GestureLeft**' & '**GestureRight**' (_as well as all the other coercion_) are now treated correctly in the simulation?

Yes it works correctly! Also in version `3.8.8` of GestureManager you can view the values of parameters controlled by animations. And yes I'm speaking of these kind of parameters:

![image](https://github.com/BlackStartx/VRC-Gesture-Manager/assets/7110292/92949491-b80e-4a8b-8688-6c593b06f55c)

**Comment by BlackStartx** [comment]
Hey Antarcticsiclepop~ ✨ 

[Gesture Manager 3.9.1](https://github.com/BlackStartx/VRC-Gesture-Manager/releases/tag/v3.9.1) is out, and it also reverts the logic of the debug view brought with 3.8.9 & 3.9.0

It should fix the issue of Animation-Clip driven parameters! 👍🏻 
Sorry for it, I'll close this issue once the 3.9.1 is merged into the curated listing!
